regression/cbmc/pointer-overflow3/test.desc

```diff
@@ -0,0 +1,16 @@
+CORE
+main.c
+--pointer-overflow-check --no-simplify
```

Comment:
At some point are we stretching the intuitive interpretation of 'overflow' too much?

Reply:
I don't mind very much whether we should cover these under `--pointer-check` or `--pointer-overflow-check`. My preference towards `--pointer-overflow-check` stems from the following sentence in C11, 6.5.6 Additive operators: "If both the pointer operand and the result point to elements of the same array object, or one past the last element of the array object, the evaluation shall not produce an overflow; otherwise, the behavior is undefined." My thoughts were: if the arithmetic takes it out-of-bounds, then we could reasonably treat it as an "overflow." (If our pointer encoding were sound, we wouldn't even need any extra checks here: the arithmetic overflow already being covered would do the trick, as we have to assume that an object could be placed right at the boundary of (virtual) memory.)

Reply:
Fair.